src/fast_forward.rs

```diff
@@ -110,8 +112,8 @@ pub fn fast_forward_singleton(
         return Err(Error::PuzzleHashMismatch);
     }
 
-    let singleton = CurriedProgram::<SingletonArgs>::from_clvm(a, puzzle)?;
-    let mut new_solution = SingletonSolution::from_clvm(a, solution)?;
+    let singleton: CurriedProgram<NodePtr, SingletonArgs<NodePtr>> = FromPtr::from_ptr(a, puzzle)?;
```

Comment:
why do you need the `NodePtr` template argument here? isn't that implied by `FromPtr`?

Reply:
You would think. I think the Rust compiler has limitations on what it can infer